src/Microsoft.Data.SqlClient/tests/UnitTests/PoolTests.cs

```diff
@@ -0,0 +1,539 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET8_0_OR_GREATER
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.NetCore.UnitTests.Util;
+using Microsoft.Data.SqlClientX;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.NetCore.UnitTests.Handlers.Prelogin
+{
+    public class PoolTests
+    {
+        private TestBase testBase;
+
+        public PoolTests()
+        {
+            testBase = new TestBase();
+        }
+
+
+        [Fact]
+        public async Task MinPoolSize_equals_MaxPoolSize()
+        {
+            await using var dataSource = testBase.CreateDataSource(csb =>
+            {
+                csb.MinPoolSize = 30;
+                csb.MaxPoolSize = 30;
+            });
+            await using var conn = await dataSource.OpenConnectionAsync();
+        }
+
+        [Fact]
+        public void MinPoolSize_bigger_than_MaxPoolSize_throws()
+            => Assert.ThrowsAsync<ArgumentException>(async () =>
+            {
+                await using var dataSource = testBase.CreateDataSource(csb =>
+                {
+                    csb.MinPoolSize = 2;
+                    csb.MaxPoolSize = 1;
+                });
+            });
+
+        [Fact]
+        public async Task Reuse_connector_before_creating_new()
+        {
+            await using var dataSource = testBase.CreateDataSource();
+            await using var conn = await dataSource.OpenConnectionAsync();
+            var backendId = conn.InternalConnection!.ServerProcessId;
+            Debug.WriteLine($"server process id: {backendId}");
+            await conn.CloseAsync();
+            await conn.OpenAsync();
+            Assert.Equal(conn.InternalConnection!.ServerProcessId, backendId);
+        }
+
+        public static readonly object[][] AsyncParams = new object[][] { 
+            new object[] { true }, 
+            new object[] { false } 
+        };
+
+        [Theory]
+        [MemberData(nameof(AsyncParams))]
```

Comment:
For situations like this (compile-time constant parameters) you can use `[InlineData(true)]` and `[InlineData(false)]` rather than set up the async params.

However, I'd personally recommend creating two different tests for async and sync. Since they're two different methods being testes, it makes sense to me to split into two tests rather than switch between the two different methods in the same test.